### ISSUE TITLE
ZIO 2.0: suggestion to have fallback ZLayer error messages with ASCII characters for Windows users

### ISSUE THREAD

**Issue by alphaho** [zio/zio#6225]
By default, Windows is not using UTF-8 encoding in the console.
And when we have the ZLayer errors, the nice error message will be rendered in unreadable codes, like below:

![image](https://user-images.githubusercontent.com/666504/147526977-92f1415a-ef71-4f40-aadd-7a2c74e730aa.png)

So far, I have noticed unreadable codes from two places. One is the "ZLayer ERROR" as shown above, another is the ZLayer Debug tree, that would print below dependency tree
```
[info]   ZLayer Wiring Graph
[info]
[info] ◉ OldLady.live
[info] ├─◑ Spider.live
[info] │ ╰─◑ Fly.live
[info] │   ╰─◑ Console.live
[info] ├─◑ Bear.live
[info] │ ╰─◑ Fly.live
[info] │   ╰─◑ Console.live
[info] ╰─◑ Console.live
[info]
[info] ◉ Console.live
[info]
[info]     ZLayer.make[OldLady with Console](
[info]                                      ^
```

like this:

![image2](https://user-images.githubusercontent.com/666504/147529765-25aec4e1-620b-4402-ab63-b54ba1d4b9dd.png)

It may be better if we can fallback to use ASCII characters when it's running in Windows(or even better, non-UTF-8 platforms). Or it would be better to help point people to below instructions ([reference](https://akr.am/blog/posts/using-utf-8-in-the-windows-terminal))

> Enable the new UTF-8 option in Windows settings. Go to the language settings, click Administrative language settings, then Change system locale… and tick the Beta: Use Unicode UTF-8 for worldwide language support option.

I tried updating the above setting in Windows, and it now can print UTF-8 correctly.

**Comment by adamgfraser** [comment]
Yes. I wonder if it is better to do everything with ASCII characters even though it may not be as artistic but one way or another these messages should definitely be readable for users regardless of platform.

**Comment by alphaho** [comment]
Personally, I really like the nice looking UTF-8 characters. I think it's more expressive than the traditional ASCII ones(examples attached below).
But I'm not sure whether we can tell in the runtime that the platform supports UTF-8 or not.

Here's the same tree rendered in ASCII:

```
OldLady.live
├── Spider.live
│   └── Fly.live
│       └── Console.live
└── Bear.live
    ├── Fly.live
    │   └── Console.live
    └── Console.live
```

While it originally would be:

```
◉ OldLady.live
├─◑ Spider.live
│ ╰─◑ Fly.live
│   ╰─◑ Console.live
├─◑ Bear.live
│ ╰─◑ Fly.live
│   ╰─◑ Console.live
╰─◑ Console.live
```

**Comment by YulawOne** [comment]
I personally in favour of UTF-8 only and add a hint to documentation with an instruction on how to enable UTF-8 support in Windows. New Windows versions will be coming with a new Terminal environment that works UTF-8 by default and most of professional developers on Windows probably already use modern environment. Therefore, fallback to ASCII won't be suitable for everyone (as I'm not sure if it possible reliably to detect UTF-8 support) and might create additional complication to the code, unless everything by default will be in ASCII, which might reduce expressiveness

**Comment by fsvehla** [comment]
@alphaho Can you post the content of your `TERM` environment variable?
Checking it should be sufficient to check the most suitable rendering.

**Comment by alphaho** [comment]
@fsvehla As it's on widows, `TERM` environment variable is not defined.

![](https://user-images.githubusercontent.com/666504/154790498-00d302a3-e7e4-4792-bbe2-a26c0ceeb195.png)
